File: code/code_c/parallel.c
#include<omp.h>
#include<stdio.h>
#include<stdlib.h>
#include <stdbool.h>

int main(int argc, char *argv[])
{
  double a=0;
  int rang;
  bool para;
  a=92.48;para=false;
  #pragma omp parallel private(rang,para)
  {
    rang=omp_get_thread_num();
    para=omp_in_parallel();
    printf("a vaut %f rang = %d para = %d\n",a,rang,para);
  }
}

File: code/parallel.f90
Program premier

	 !$ use OMP_LIB
    Implicit none

    real :: a,b
    logical :: para
    Integer :: rang
    a=92.48
    para=.False.

    !$OMP Parallel Private(rang,para)
    !$ rang=OMP_GET_THREAD_NUM()
    !$ para=OMP_IN_PARALLEL()

    print*, "a vaut", a, "rang = ", rang, " para = ", para
    ! Export OMP_GET_THREADS=6
    !$OMP END PARALLEL
end program premier
